rohmu/__init__.py

```diff
@@ -9,45 +9,45 @@
 from .errors import InvalidConfigurationError
 from .notifier.interface import Notifier
 from .object_storage.base import BaseTransfer
-from typing import Any, Dict, Mapping, Type
+from typing import Any, cast, Dict, Type
 
 IO_BLOCK_SIZE = 2**20  # 1 MiB
 STORAGE_TYPE = "storage_type"
 NOTIFIER_TYPE = "notifier_type"
-Config = Mapping[str, Any]
+Config = Dict[str, Any]
 
 
-def get_class_for_transfer(obj_store: Config) -> Type[BaseTransfer]:
+def get_class_for_transfer(obj_store: Config) -> Type[BaseTransfer[StorageModel]]:
```

Comment:
Have you considered using "tagged unions" for typing these transfer objects? 
See https://docs.pydantic.dev/latest/usage/types/#discriminated-unions-aka-tagged-unions
It seems better than casting to a "base" type

Reply:
That seems like work for a different PR IMHO... Here I just fixed the type to use stricter mypy settings which was the goal of this PR. What you suggest seems to require some changes in the class hierarchy to make it work.